icu4c/source/common/utrie2_builder.cpp

```diff
@@ -309,13 +309,16 @@ utrie2_clone(const UTrie2 *other, UErrorCode *pErrorCode) {
 
     trie=(UTrie2 *)uprv_malloc(sizeof(UTrie2));
     if(trie==NULL) {
+        *pErrorCode=U_MEMORY_ALLOCATION_ERROR;
         return NULL;
     }
     uprv_memcpy(trie, other, sizeof(UTrie2));
 
     if(other->memory!=NULL) {
         trie->memory=uprv_malloc(other->length);
-        if(trie->memory!=NULL) {
+        if(trie->memory==NULL) {
+            *pErrorCode=U_MEMORY_ALLOCATION_ERROR;
+        } else {
```

Comment:
At line 335, the call to cloneBuilder() can also return null, and needs to be checked.
(annoying that github won't let me add a review comment there) 
You could maybe combine the two checks by setting the errorCode at line 339.

Reply:
Thank you Andy for taking a look at the PR!
I just realized that I accidentally made my commit to jefgen:master instead of a separate branch. However, I don't think I can change the branch on an existing PR. I'll need to close this PR and open another one (with your suggested changes).
Thanks again!